source/core/TDCodecVia.cpp

```diff
@@ -2329,6 +2329,56 @@ void TDViaFormatter::FormatClusterData(TypeRef type, void *pData)
     _string->Append(Fmt()._clusterPost);
 }
 //------------------------------------------------------------
+void TDViaFormatter::FormatVariant(TypeRef type, void *pData)
+{
+    _string->Append(Fmt()._clusterPre);
+    Boolean useQuotes = Fmt().QuoteFieldNames();
+    VariantDataRef variantData = *reinterpret_cast<VariantDataRef *>(pData);
+
+    if (useQuotes)
+        _string->Append('\"');
+    _string->AppendCStr(variantInnerData);
+    if (useQuotes)
+        _string->Append('\"');
+    _string->Append(*tsNameSuffix);
+
+    TypeRef underlyingType = variantData->GetInnerType();
+    if (underlyingType) {
+        FormatData(underlyingType, underlyingType->Begin(kPARead));
+    } else {
+        _string->AppendCStr("null");
+    }
+    _string->Append(Fmt()._itemSeparator);
+    if (useQuotes)
+        _string->Append('\"');
+    _string->AppendCStr(variantAttributes);
+    if (useQuotes)
+        _string->Append('\"');
+    _string->Append(*tsNameSuffix);
+    if (variantData->HasMap()) {
+        _string->Append(Fmt()._clusterPre);
+        auto iterator = variantData->GetMap_cbegin(), iteratorBegin = variantData->GetMap_cbegin(), iteratorEnd = variantData->GetMap_cend();
+        for (; iterator != iteratorEnd; ++iterator) {
+            if (iterator != iteratorBegin) {
+                _string->Append(Fmt()._itemSeparator);
+            }
+            StringRef attributeName = iterator->first;
+            if (useQuotes)
```

Comment:
Can you do the pattern shown here?: https://github.com/ni/VireoSDK/blob/669a35df79af13ccac2c3f3039bc54841b5fbfca/source/core/TDCodecVia.cpp#L2235

Then we can avoid hardcoding the quote character and make sure to escape the attribute name string

Reply:
Done.